lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -305,10 +352,41 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		// we will call the specialized coin selection function for
 		// that.
 		case r.FundUpToMaxAmt != 0 && r.MinFundAmt != 0:
+
```

Comment:
I think we can simplify this a bit by adding a new case statement here for when a set of coins is selected manually. From there, we just need to validate that we know them all (should fail above if not) and that they sum up to the proper count.

Reply:
Is the idea that users must also specify a max funding amount to signal that all should be spent?

Reply:
I updated the description of this PR to clarify how I envision this to work. Basically, when opening a channel, you can specify if you want to take your entire wallet balance(default) or the sum of selected UTXOs as basis for the funding flow. Then you can further specify if you want to use a fraction of the selected funds to go towards a channel opening via `local_amt` or if you want the all of it consumed by the opening via `fundmax`.

The `r.FundUpToMaxAmt != 0 && r.MinFundAmt != 0` is a fragment from the `fundmax` PR. Only if both parameters are set do we take it as indication that the `fundmax` flow is executing. Otherwise someone could pass in `r.FundUpToMaxAmt` expecting this to execute, but `r.MinFundAmt` is crucial as well.